modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -311,4 +332,42 @@ private static ReadOnlyObjectProperty<? extends Control> getControlProperty(Obje
 
         return null;
     }
+
+    // We need to store all the listeners added to each ObservableList so that we can remove them. For this we need
+    // a map mapping ObservableList to various listeners such as ListChangeListeners, but the map needs to be a
+    // WeakHashMap.
+    // The ideal key to the WeakHashMap would be the ObservableList itself, except for the fact that its equals method
+    // compares the list contents. If a WeakIdentityHashMap existed, we could use that instead with the ObservableList
+    // as the key.
+    // We can't use an IdentityWrapper as the key to the HashMap because we need a strong reference (ideally from the
+    // ObservableList itself) to the IdentityWrapper else it'll be garbage collected.
+    // Since every ObservableList gets a ListChangeListener, we can use that as an IdentityWrapper and rely on the fact
+    // ObservableList has a strong reference to the ListChangeListener.
+    static class IdentityWrapperListChangeListener implements ListChangeListener<MenuItem> {
+
+        ObservableList<? extends MenuItem> innerList;
+
+        public IdentityWrapperListChangeListener(ObservableList<? extends MenuItem> list) {
+            this.innerList = list;
+        }
+
+        @Override
+        public void onChanged(Change<? extends MenuItem> c) {
```

Comment:
the use of this class as both key and value (the latter overrides this method) was a bit confusing to me.  clever, but it took some time to understand.  Would it be possible to explain in the comments here and on lines 146 and 160 what is going on?